src/views/lighttable.c

```diff
@@ -616,7 +626,71 @@ static int expose_filemanager(dt_view_t *self, cairo_t *cr, int32_t width, int32
         if(iir == 1 && row) continue;
 
         /* set mouse over id if pointer is in current row / col */
-        if(pi == col && pj == row)
+        if (lib->using_arrows)
+        {
+
+          if(before_mouse_over_id == -1)
+          {
+            // mouse has never been in filemanager area set mouse on first image and ignore this movement
+            before_mouse_over_id = query_ids[0];
+          }
+          if(before_mouse_over_id == id)
+          {
+            // I would like to jump from before_mouse_over_id to query_ids[idx]
+            int idx = current_image+lib->key_jump_offset-1;
+            int current_row = (int)((current_image-1)/iir);
+            int current_col = current_image%iir;
+
+            // detect if the current movement need some extra movement (page adjust)
+            if (current_row  == (int)(max_rows-1.5) && lib->key_jump_offset == iir)
+              // going DOWN from last row
+              move_view(lib,DOWN);
+            else if (current_row  == 0 && lib->key_jump_offset == iir*-1)
+            {
+              // going UP from first row
+              move_view(lib,UP);
+            }
+            else if (current_row == (int)(max_rows-1.5) && current_col ==  0 && lib->key_jump_offset == 1)
+              // going RIGHT from last visible
+              move_view(lib,DOWN);
+            else if (current_row == 0 && current_col ==  1 && lib->key_jump_offset == -1)
+              // going LEFT from first visible
+              move_view(lib,UP);
+            if (idx > -1 && idx < lib->collection_count && query_ids[idx])
```

Comment:
@johnnyrun 
I highly suspect that `lib->collection_count` is wrong here and in other places:

```
=================================================================
==29685==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x6040005f6e4c at pc 0x7fffdbf441de bp 0x7fffffffc660 sp 0x7fffffffc650
READ of size 4 at 0x6040005f6e4c thread T0
    #0 0x7fffdbf441dd in expose_filemanager /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:719
    #1 0x7fffdbf441dd in expose /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:1418
    #2 0x7ffff69c2f72 in dt_view_manager_expose /tmp/makepkg/darktable-git/src/darktable/src/views/view.c:521
    #3 0x7ffff685d3ed in dt_control_expose /tmp/makepkg/darktable-git/src/darktable/src/control/control.c:297
    #4 0x7ffff697cd26 in draw /tmp/makepkg/darktable-git/src/darktable/src/gui/gtk.c:447
    #5 0x7ffff4a48830  (/usr/lib/libgtk-3.so.0+0x222830)
    #6 0x7ffff4b885fe  (/usr/lib/libgtk-3.so.0+0x3625fe)
    #7 0x7ffff35d91b3  (/usr/lib/libgobject-2.0.so.0+0x101b3)
    #8 0x7ffff35f33c0 in g_signal_emit_valist (/usr/lib/libgobject-2.0.so.0+0x2a3c0)
    #9 0x7ffff35f3fde in g_signal_emit (/usr/lib/libgobject-2.0.so.0+0x2afde)
    #10 0x7ffff4b951b9  (/usr/lib/libgtk-3.so.0+0x36f1b9)
    #11 0x7ffff498823f in gtk_container_propagate_draw (/usr/lib/libgtk-3.so.0+0x16223f)
    #12 0x7ffff4988321  (/usr/lib/libgtk-3.so.0+0x162321)
    #13 0x7ffff493cf83  (/usr/lib/libgtk-3.so.0+0x116f83)
    #14 0x7ffff498d2ec  (/usr/lib/libgtk-3.so.0+0x1672ec)
    #15 0x7ffff4991f4f  (/usr/lib/libgtk-3.so.0+0x16bf4f)
    #16 0x7ffff493f920  (/usr/lib/libgtk-3.so.0+0x119920)
    #17 0x7ffff4b94fda  (/usr/lib/libgtk-3.so.0+0x36efda)
    #18 0x7ffff498823f in gtk_container_propagate_draw (/usr/lib/libgtk-3.so.0+0x16223f)
    #19 0x7ffff4988321  (/usr/lib/libgtk-3.so.0+0x162321)
    #20 0x7ffff4a05503  (/usr/lib/libgtk-3.so.0+0x1df503)
    #21 0x7ffff498d2ec  (/usr/lib/libgtk-3.so.0+0x1672ec)
    #22 0x7ffff4991f4f  (/usr/lib/libgtk-3.so.0+0x16bf4f)
    #23 0x7ffff4a06510  (/usr/lib/libgtk-3.so.0+0x1e0510)
    #24 0x7ffff4b94fda  (/usr/lib/libgtk-3.so.0+0x36efda)
    #25 0x7ffff498823f in gtk_container_propagate_draw (/usr/lib/libgtk-3.so.0+0x16223f)
    #26 0x7ffff4988321  (/usr/lib/libgtk-3.so.0+0x162321)
    #27 0x7ffff493cf83  (/usr/lib/libgtk-3.so.0+0x116f83)
    #28 0x7ffff498d2ec  (/usr/lib/libgtk-3.so.0+0x1672ec)
    #29 0x7ffff4991f4f  (/usr/lib/libgtk-3.so.0+0x16bf4f)
    #30 0x7ffff493f920  (/usr/lib/libgtk-3.so.0+0x119920)
    #31 0x7ffff4b94fda  (/usr/lib/libgtk-3.so.0+0x36efda)
    #32 0x7ffff498823f in gtk_container_propagate_draw (/usr/lib/libgtk-3.so.0+0x16223f)
    #33 0x7ffff4988321  (/usr/lib/libgtk-3.so.0+0x162321)
    #34 0x7ffff4ba2940  (/usr/lib/libgtk-3.so.0+0x37c940)
    #35 0x7ffff4b94fda  (/usr/lib/libgtk-3.so.0+0x36efda)
    #36 0x7ffff4b954b8 in gtk_widget_send_expose (/usr/lib/libgtk-3.so.0+0x36f4b8)
    #37 0x7ffff4a478a4 in gtk_main_do_event (/usr/lib/libgtk-3.so.0+0x2218a4)
    #38 0x7ffff457e754  (/usr/lib/libgdk-3.so.0+0x30754)
    #39 0x7ffff458d447  (/usr/lib/libgdk-3.so.0+0x3f447)
    #40 0x7ffff458e59b  (/usr/lib/libgdk-3.so.0+0x4059b)
    #41 0x7ffff458e752  (/usr/lib/libgdk-3.so.0+0x40752)
    #42 0x7ffff35d91b3  (/usr/lib/libgobject-2.0.so.0+0x101b3)
    #43 0x7ffff35f38ec in g_signal_emit_valist (/usr/lib/libgobject-2.0.so.0+0x2a8ec)
    #44 0x7ffff35f3fde in g_signal_emit (/usr/lib/libgobject-2.0.so.0+0x2afde)
    #45 0x7ffff45866e0  (/usr/lib/libgdk-3.so.0+0x386e0)
    #46 0x7ffff4574ab7  (/usr/lib/libgdk-3.so.0+0x26ab7)
    #47 0x7ffff5151792  (/usr/lib/libglib-2.0.so.0+0x4a792)
    #48 0x7ffff5150d19 in g_main_context_dispatch (/usr/lib/libglib-2.0.so.0+0x49d19)
    #49 0x7ffff51510cf  (/usr/lib/libglib-2.0.so.0+0x4a0cf)
    #50 0x7ffff51513f1 in g_main_loop_run (/usr/lib/libglib-2.0.so.0+0x4a3f1)
    #51 0x7ffff4a46a64 in gtk_main (/usr/lib/libgtk-3.so.0+0x220a64)
    #52 0x7ffff6982664 in dt_gui_gtk_run /tmp/makepkg/darktable-git/src/darktable/src/gui/gtk.c:981
    #53 0x400890 in main /tmp/makepkg/darktable-git/src/darktable/src/main.c:25
    #54 0x7ffff60e5290 in __libc_start_main (/usr/lib/libc.so.6+0x20290)
    #55 0x4008e9 in _start (/usr/bin/darktable+0x4008e9)

0x6040005f6e4c is located 4 bytes to the left of 36-byte region [0x6040005f6e50,0x6040005f6e74)
allocated by thread T0 here:
    #0 0x7ffff6eff020 in __interceptor_calloc /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cc:70
    #1 0x7fffdbf3fc0a in expose_filemanager /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:616
    #2 0x7fffdbf3fc0a in expose /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:1418
    #3 0x7ffff69c2f72 in dt_view_manager_expose /tmp/makepkg/darktable-git/src/darktable/src/views/view.c:521
    #4 0x7ffff685d3ed in dt_control_expose /tmp/makepkg/darktable-git/src/darktable/src/control/control.c:297
    #5 0x7ffff697cd26 in draw /tmp/makepkg/darktable-git/src/darktable/src/gui/gtk.c:447
    #6 0x7ffff4a48830  (/usr/lib/libgtk-3.so.0+0x222830)

SUMMARY: AddressSanitizer: heap-buffer-overflow /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:719 in expose_filemanager
Shadow bytes around the buggy address:
  0x0c08800b6d70: fa fa fa fa fa fa fa fa fa fa fd fd fd fd fd fa
  0x0c08800b6d80: fa fa fd fd fd fd fd fa fa fa fd fd fd fd fd fd
  0x0c08800b6d90: fa fa fd fd fd fd fd fa fa fa fd fd fd fd fd fa
  0x0c08800b6da0: fa fa fd fd fd fd fd fa fa fa fd fd fd fd fd fd
  0x0c08800b6db0: fa fa fd fd fd fd fd fd fa fa fd fd fd fd fd fd
=>0x0c08800b6dc0: fa fa fd fd fd fd fd fa fa[fa]00 00 00 00 04 fa
  0x0c08800b6dd0: fa fa fd fd fd fd fd fa fa fa fd fd fd fd fd fd
  0x0c08800b6de0: fa fa fd fd fd fd fd fa fa fa fd fd fd fd fd fa
  0x0c08800b6df0: fa fa fd fd fd fd fd fa fa fa fd fd fd fd fd fa
  0x0c08800b6e00: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c08800b6e10: fa fa fa fa fa fa fa fa fa fa fd fd fd fd fd fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Heap right redzone:      fb
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack partial redzone:   f4
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==29685==ABORTING
```

and 

```
=================================================================
==30312==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x6040006659b4 at pc 0x7fffdbf404fb bp 0x7fffffffc660 sp 0x7fffffffc650
READ of size 4 at 0x6040006659b4 thread T0
    #0 0x7fffdbf404fa in expose_filemanager /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:693
    #1 0x7fffdbf404fa in expose /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:1418
    #2 0x7ffff69c2f72 in dt_view_manager_expose /tmp/makepkg/darktable-git/src/darktable/src/views/view.c:521
    #3 0x7ffff685d3ed in dt_control_expose /tmp/makepkg/darktable-git/src/darktable/src/control/control.c:297
    #4 0x7ffff697cd26 in draw /tmp/makepkg/darktable-git/src/darktable/src/gui/gtk.c:447
    #5 0x7ffff4a48830  (/usr/lib/libgtk-3.so.0+0x222830)
    #6 0x7ffff4b885fe  (/usr/lib/libgtk-3.so.0+0x3625fe)
    #7 0x7ffff35d91b3  (/usr/lib/libgobject-2.0.so.0+0x101b3)
    #8 0x7ffff35f33c0 in g_signal_emit_valist (/usr/lib/libgobject-2.0.so.0+0x2a3c0)
    #9 0x7ffff35f3fde in g_signal_emit (/usr/lib/libgobject-2.0.so.0+0x2afde)
    #10 0x7ffff4b951b9  (/usr/lib/libgtk-3.so.0+0x36f1b9)
    #11 0x7ffff498823f in gtk_container_propagate_draw (/usr/lib/libgtk-3.so.0+0x16223f)
    #12 0x7ffff4988321  (/usr/lib/libgtk-3.so.0+0x162321)
    #13 0x7ffff493cf83  (/usr/lib/libgtk-3.so.0+0x116f83)
    #14 0x7ffff498d2ec  (/usr/lib/libgtk-3.so.0+0x1672ec)
    #15 0x7ffff4991f4f  (/usr/lib/libgtk-3.so.0+0x16bf4f)
    #16 0x7ffff493f920  (/usr/lib/libgtk-3.so.0+0x119920)
    #17 0x7ffff4b94fda  (/usr/lib/libgtk-3.so.0+0x36efda)
    #18 0x7ffff498823f in gtk_container_propagate_draw (/usr/lib/libgtk-3.so.0+0x16223f)
    #19 0x7ffff4988321  (/usr/lib/libgtk-3.so.0+0x162321)
    #20 0x7ffff4a05503  (/usr/lib/libgtk-3.so.0+0x1df503)
    #21 0x7ffff498d2ec  (/usr/lib/libgtk-3.so.0+0x1672ec)
    #22 0x7ffff4991f4f  (/usr/lib/libgtk-3.so.0+0x16bf4f)
    #23 0x7ffff4a06510  (/usr/lib/libgtk-3.so.0+0x1e0510)
    #24 0x7ffff4b94fda  (/usr/lib/libgtk-3.so.0+0x36efda)
    #25 0x7ffff498823f in gtk_container_propagate_draw (/usr/lib/libgtk-3.so.0+0x16223f)
    #26 0x7ffff4988321  (/usr/lib/libgtk-3.so.0+0x162321)
    #27 0x7ffff493cf83  (/usr/lib/libgtk-3.so.0+0x116f83)
    #28 0x7ffff498d2ec  (/usr/lib/libgtk-3.so.0+0x1672ec)
    #29 0x7ffff4991f4f  (/usr/lib/libgtk-3.so.0+0x16bf4f)
    #30 0x7ffff493f920  (/usr/lib/libgtk-3.so.0+0x119920)
    #31 0x7ffff4b94fda  (/usr/lib/libgtk-3.so.0+0x36efda)
    #32 0x7ffff498823f in gtk_container_propagate_draw (/usr/lib/libgtk-3.so.0+0x16223f)
    #33 0x7ffff4988321  (/usr/lib/libgtk-3.so.0+0x162321)
    #34 0x7ffff4ba2940  (/usr/lib/libgtk-3.so.0+0x37c940)
    #35 0x7ffff4b94fda  (/usr/lib/libgtk-3.so.0+0x36efda)
    #36 0x7ffff4b954b8 in gtk_widget_send_expose (/usr/lib/libgtk-3.so.0+0x36f4b8)
    #37 0x7ffff4a478a4 in gtk_main_do_event (/usr/lib/libgtk-3.so.0+0x2218a4)
    #38 0x7ffff457e754  (/usr/lib/libgdk-3.so.0+0x30754)
    #39 0x7ffff458d447  (/usr/lib/libgdk-3.so.0+0x3f447)
    #40 0x7ffff458e59b  (/usr/lib/libgdk-3.so.0+0x4059b)
    #41 0x7ffff458e752  (/usr/lib/libgdk-3.so.0+0x40752)
    #42 0x7ffff35d91b3  (/usr/lib/libgobject-2.0.so.0+0x101b3)
    #43 0x7ffff35f38ec in g_signal_emit_valist (/usr/lib/libgobject-2.0.so.0+0x2a8ec)
    #44 0x7ffff35f3fde in g_signal_emit (/usr/lib/libgobject-2.0.so.0+0x2afde)
    #45 0x7ffff45866e0  (/usr/lib/libgdk-3.so.0+0x386e0)
    #46 0x7ffff4574ab7  (/usr/lib/libgdk-3.so.0+0x26ab7)
    #47 0x7ffff5151792  (/usr/lib/libglib-2.0.so.0+0x4a792)
    #48 0x7ffff5150d19 in g_main_context_dispatch (/usr/lib/libglib-2.0.so.0+0x49d19)
    #49 0x7ffff51510cf  (/usr/lib/libglib-2.0.so.0+0x4a0cf)
    #50 0x7ffff51513f1 in g_main_loop_run (/usr/lib/libglib-2.0.so.0+0x4a3f1)
    #51 0x7ffff4a46a64 in gtk_main (/usr/lib/libgtk-3.so.0+0x220a64)
    #52 0x7ffff6982664 in dt_gui_gtk_run /tmp/makepkg/darktable-git/src/darktable/src/gui/gtk.c:981
    #53 0x400890 in main /tmp/makepkg/darktable-git/src/darktable/src/main.c:25
    #54 0x7ffff60e5290 in __libc_start_main (/usr/lib/libc.so.6+0x20290)
    #55 0x4008e9 in _start (/usr/bin/darktable+0x4008e9)

0x6040006659b4 is located 0 bytes to the right of 36-byte region [0x604000665990,0x6040006659b4)
allocated by thread T0 here:
    #0 0x7ffff6eff020 in __interceptor_calloc /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cc:70
    #1 0x7fffdbf3fc0a in expose_filemanager /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:616
    #2 0x7fffdbf3fc0a in expose /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:1418
    #3 0x7ffff69c2f72 in dt_view_manager_expose /tmp/makepkg/darktable-git/src/darktable/src/views/view.c:521
    #4 0x7ffff685d3ed in dt_control_expose /tmp/makepkg/darktable-git/src/darktable/src/control/control.c:297
    #5 0x7ffff697cd26 in draw /tmp/makepkg/darktable-git/src/darktable/src/gui/gtk.c:447
    #6 0x7ffff4a48830  (/usr/lib/libgtk-3.so.0+0x222830)

SUMMARY: AddressSanitizer: heap-buffer-overflow /tmp/makepkg/darktable-git/src/darktable/src/views/lighttable.c:693 in expose_filemanager
Shadow bytes around the buggy address:
  0x0c08800c4ae0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c08800c4af0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c08800c4b00: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c08800c4b10: fa fa fa fa fa fa fa fa fa fa 00 00 00 00 00 00
  0x0c08800c4b20: fa fa fa fa fa fa fa fa fa fa 00 00 00 00 00 00
=>0x0c08800c4b30: fa fa 00 00 00 00[04]fa fa fa fd fd fd fd fd fa
  0x0c08800c4b40: fa fa fd fd fd fd fd fa fa fa fa fa fa fa fa fa
  0x0c08800c4b50: fa fa 00 00 00 00 00 00 fa fa fa fa fa fa fa fa
  0x0c08800c4b60: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c08800c4b70: fa fa fa fa fa fa fa fa fa fa 00 00 00 00 00 fa
  0x0c08800c4b80: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Heap right redzone:      fb
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack partial redzone:   f4
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==30312==ABORTING
```